src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -157,17 +157,21 @@ public static string GetFileNameWithoutExtension(string path)
 
         // Returns a cryptographically strong random 8.3 string that can be 
         // used as either a folder name or a file name.
-        public static string GetRandomFileName()
+        public static unsafe string GetRandomFileName()
         {
-            // 5 bytes == 40 bits == 40/5 == 8 chars in our encoding.
-            // So 10 bytes provides 16 chars, of which we need 11
-            // for the 8.3 name.
-            byte[] key = CreateCryptoRandomByteArray(10);
-
-            // rndCharArray is expected to be 16 chars
-            char[] rndCharArray = ToBase32StringSuitableForDirName(key).ToCharArray();
-            rndCharArray[8] = '.';
-            return new string(rndCharArray, 0, 12);
+            // 9 random bytes provides 12 chars in our encoding for the 8.3 name.
+            const int KeyLength = 9;
+            byte* pKey = stackalloc byte[KeyLength];
+            GetCryptoRandomBytes(pKey, KeyLength);
+
+            const int RandomFileNameLength = 12;
+            string randomFileName = new string('\0', RandomFileNameLength);
+            fixed (char* pRandomFileName = randomFileName)
```

Comment:
FWIW, I would also propose the stackalloc thing.  I understand that there's an addition copy when you create the string, but unless performance traces showed that copy was killing us, I'd rather write the "safer" version of the code, just to encourage good habits.